### PR TITLE
 Avoid mutation of original bullet points

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: ONE-5062
---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/979/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4002/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/979/ : UNSTABLE in 14m 32s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4001/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5639/ : SUCCESS in 19m 10s

**Comment by ivan-nejezchleb** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1442

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4006/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/188/ : SUCCESS in 9m 03s

**Comment by ivan-nejezchleb** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1442

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4007/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/189/ : SUCCESS in 7m 04s

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/980/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4009/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/980/ : UNSTABLE in 15m 50s

**Review by patrikbraborec — APPROVED** [review]

File: libs/sdk-ui-charts/src/highcharts/chartTypes/_chartCreators/setupDrilldownToParentAttribute.ts [hunk 1]
```diff
@@ -17,13 +17,14 @@ export function getDDPointsInParentTick(axis: any, tick: IHighchartsParentTick):
     }
 
     // replace y value by target value for bullet chart target
-    ddPoints.forEach((ddPoint) => {
+    const modifiedPoints = ddPoints.map((ddPoint) => {
         if ((ddPoint.series as any)?.userOptions?.bulletChartMeasureType === "target") {
-            ddPoint.y = ddPoint.isNullTarget ? null : ddPoint.target;
+            return Object.assign({}, ddPoint, { y: ddPoint.isNullTarget ? null : ddPoint.target });
```
Comment (pbenes) [inline libs/sdk-ui-charts/src/highcharts/chartTypes/_chartCreators/setupDrilldownToParentAttribute.ts:22]: maybe I'm missing something, but what would you think about `return { ...ddPoint, y: ddPoint.isNullTarget ? null : ddPoint.target }`

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1442

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

File: libs/sdk-ui-charts/src/highcharts/chartTypes/_chartCreators/setupDrilldownToParentAttribute.ts (hunk 1 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-charts/src/highcharts/chartTypes/_chartCreators/setupDrilldownToParentAttribute.ts:22]: ddPoint is class instance with some methods and spread operator is not working on it

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1170/ : SUCCESS in 1m 06s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1444/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/190/ : SUCCESS in 5m 47s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5646/ : SUCCESS in 15m 43s

**Pull request merged**